quic/s2n-quic-core/src/recovery/hybrid_slow_start.rs

```diff
@@ -94,29 +135,73 @@ impl HybridSlowStart {
         if let (N_SAMPLING, Some(last_min_rtt), Some(cur_min_rtt)) =
             (self.sample_count, self.last_min_rtt, self.cur_min_rtt)
         {
-            let threshold = last_min_rtt / THRESHOLD_DIVIDEND;
-            // Clamp n to the min and max thresholds
-            let threshold = threshold.min(MAX_DELAY_THRESHOLD).max(MIN_DELAY_THRESHOLD);
-            let delay_increase_is_over_threshold = cur_min_rtt >= last_min_rtt + threshold;
-            let congestion_window_is_above_minimum = congestion_window >= self.low_ssthresh();
-
-            if delay_increase_is_over_threshold && congestion_window_is_above_minimum {
-                self.threshold = congestion_window;
+            if congestion_window >= self.css_threshold {
+                self.css_count += 1;
+                if cur_min_rtt < self.css_baseline_min_rtt {
+                    // resume slow start
+                    self.css_threshold = self.threshold;
+                    self.ss_growth_divisor = 1.0;
+                    self.css_count = 0;
+                }
+                if self.css_count >= CSS_ROUNDS {
+                    // exit slow start phase
+                    self.threshold = congestion_window;
+                    self.css_threshold = f32::MAX;
+                    self.ss_growth_divisor = 1.0;
+                }
+            } else {
+                let threshold = last_min_rtt / THRESHOLD_DIVIDEND;
+                // Clamp n to the min and max thresholds
+                let threshold = threshold.min(MAX_DELAY_THRESHOLD).max(MIN_DELAY_THRESHOLD);
+                let delay_increase_is_over_threshold = cur_min_rtt >= last_min_rtt + threshold;
+                let congestion_window_is_above_minimum = congestion_window >= self.low_ssthresh();
+
+                if self.use_hystart_plus_plus {
+                    // if delay is beyond threshold, go into css phase
+                    if delay_increase_is_over_threshold {
+                        self.css_threshold = congestion_window;
+                        self.css_baseline_min_rtt = cur_min_rtt;
+                        self.ss_growth_divisor = CSS_GROWTH_DIVISOR;
+                        self.css_count = 0;
+                    }
+                } else if delay_increase_is_over_threshold && congestion_window_is_above_minimum {
+                    self.threshold = congestion_window;
+                }
             }
         }
     }
 
+    /// return cwnd increment during slow start phase
+    /// should be called from on_packet_ack
+    pub fn cwnd_increment(&self, sent_bytes: usize) -> f32 {
+        if cfg!(debug_assertions) {
```

Comment:
did this change get missed from the other PR?

Reply:
oh. sorry. I missed one commit.